lib/screens/v2/candidate_generator/on_bus/departures.ex

```diff
@@ -80,6 +91,29 @@ defmodule Screens.V2.CandidateGenerator.Widgets.OnBus.Departures do
     Enum.filter(departures, &(&1.prediction.route.id != route_id))
   end
 
+  defp filter_duplicate_routes(departures) do
+    unique_departures =
+      Enum.uniq_by(departures, fn dep ->
+        {dep.prediction.route.line.id, dep.prediction.trip.direction_id,
```

Comment:
We don't need to include `direction_id` here since headsign should be strictly "more unique" than direction (if the direction is different then the headsign will also be different).